plugins/Dashboard/javascripts/dashboardObject.js

```diff
@@ -204,8 +204,9 @@
             }, 'get');
             ajaxRequest.setCallback(
                 function () {
-                    methods.loadDashboard.apply(this, [1]);
-                    rebuildMenu();
+                    Promise.resolve(rebuildMenu()).then(function () {
```

Comment:
This doesn't work either. When a dashboard is removed, the list in widgetized dashboard isn't refreshed. I guess the problem is, that `rebuildMenu`, doesn't return a promise in widgetized mode. simply returning the `ajaxRequest.send()` at the methods end might solve this I guess.